src/helpers/augmentedReality/checkDownloadedData.js

```diff
@@ -0,0 +1,29 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+export const checkDownloadedData = async ({ downloadableData }) => {
+  let newDownloadedData = [...downloadableData];
+
+  for (let index = 0; index < newDownloadedData.length; index++) {
+    const downloadableDataItem = newDownloadedData[index];
+
+    const { id: objectId, downloadableUris } = downloadableDataItem;
+
+    for (let itemIndex = 0; itemIndex < downloadableUris.length; itemIndex++) {
+      const { id, title, type } = downloadableUris[itemIndex];
+
+      try {
+        const downloadedItem = await AsyncStorage.getItem(`${objectId}-${title}${id}.${type}`);
+
+        if (downloadedItem) {
+          const { localUris } = JSON.parse(downloadedItem);
+
+          newDownloadedData[index] = localUris && JSON.parse(downloadedItem);
```

Comment:
I've corrected the typo.

https://github.com/ikuseiGmbH/smart-village-app-app/pull/554/commits/6d10ec18f523f9020b3f36dda2d463f0d00328d5

Reply:
is `downloadedItem` an array? or why have you decided to fallback to an empty array?

Reply:
> is downloadedItem an array?

Previously it was not an array. I wanted to use the `readFromStore` helper and this helper returned me the information as an array. 

---
> or why have you decided to fallback to an empty array?

Because I tried to remove the type error.

Reply:
ok let's have a check again together before merge

Reply:
we made it more beautiful with Live Share...